### PR TITLE
Custom bone offset

### PR DESCRIPTION
**Pull request by Emyka**
Allow the user to adjust a bone position.
only the dot, the gizmo will still appear at the original position.

This can improve the UX for facial expression.

Features a list is to visualise, drop, export and import a single or all bones

### CODE REVIEW

File: Ktisis/Structs/Bones/CustomOffset.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+			var gender = actor->Customize.Gender;
+			var race = actor->Customize.Race;
+
+			if (gender == Gender.Male)
+				switch (race) {
+					case Race.Lalafell: return BodyType.LalafellMale;
+					case Race.Roegadyn:
+					case Race.Hrothgar: return BodyType.HrothgarRoegadyn;
+					default: return BodyType.TallMale;
+				} else switch (race) {
```
Comment (Fayti1703) [inline Ktisis/Structs/Bones/CustomOffset.cs:31]: This is not correctly indented.
At minimum, I'd suggest this:
```suggestion
				} 
			else 
				switch (race) {
```

But ideally, add braces around this. It's a multi-line statement, we can afford those.

File: Ktisis/Util/GuiHelpers.cs [hunk 2]
```diff
@@ -87,6 +89,19 @@ internal class GuiHelpers {
 			}
 		}
 
+		public static bool DragFloat3FillWidth(string label, bool icon, string? tooltip, ref Vector3 vector,float speed, string format) {
```
Comment (Fayti1703) [inline Ktisis/Util/GuiHelpers.cs:92]: Minor, but:
```suggestion
		public static bool DragFloat3FillWidth(string label, bool icon, string? tooltip, ref Vector3 vector, float speed, string format) {
```

File: Ktisis/Configuration.cs [hunk 3]
```diff
@@ -88,6 +88,7 @@ public class Configuration : IPluginConfiguration {
 
 		// Data memory
 		public Dictionary<string, GlamourDresser.GlamourPlate[]?>? GlamourPlateData { get; set; } = null;
+		public Dictionary<string,Dictionary<string,Vector3>> CustomBoneOffset { get; set; } = new();
```
Comment (Fayti1703) [inline Ktisis/Configuration.cs:91]:
```suggestion
		public Dictionary<string, Dictionary<string, Vector3>> CustomBoneOffset { get; set; } = new();
```

File: Ktisis/Interface/Windows/ConfigGui.cs [hunk 4]
```diff
@@ -309,7 +313,7 @@ internal static class ConfigGui {
 		}
 		private static string PrettyKeys(List<VirtualKey>  keys) => string.Join(" + ", keys.Select(k => VirtualKeyExtensions.GetFancyName(k)));
 
-		public static void DrawDataTab(Configuration cfg) {
+		public static unsafe void DrawDataTab(Configuration cfg) {
```
Comment (Fayti1703) [inline Ktisis/Interface/Windows/ConfigGui.cs:316]: This doesn't need to be made unsafe.

**Review by Fayti1703 — COMMENTED** [review]
Found some minor formatting problems. Looks good otherwise!

File: Ktisis/Structs/Bones/CustomOffset.cs (hunk 1 above)
Comment (Emyka) [reply Ktisis/Structs/Bones/CustomOffset.cs:31]: VS offers this, i'm not sure if it's worth.
```suggestion
			else return race switch {
				Race.Lalafell => BodyType.LalafellFemale,
				_ => BodyType.TallFemale,
			};
```

Comment (Fayti1703) [reply Ktisis/Structs/Bones/CustomOffset.cs:31]: I would still break the line there. Could do both?

Comment (Emyka) [reply Ktisis/Structs/Bones/CustomOffset.cs]: Yes I noticed it was unrelated with the else being on the wrong indent (fixed with 1acc264).
I was curious of your opinion about "switch statement expression" style offered by vs.

Comment (Fayti1703) [reply Ktisis/Structs/Bones/CustomOffset.cs]: Switch expressions are neat, imo. Don't see a reason not to use them.
(Though if you apply it to the `else` branch, you should probably apply it to the `if` branch as well, for consistency)

**Pull request merged**